syncclient/client.py

```diff
@@ -109,6 +109,15 @@ def _request(self, method, url, **kwargs):
         url = self.api_endpoint.rstrip('/') + '/' + url.lstrip('/')
         self.raw_resp = requests.request(method, url, auth=self.auth, **kwargs)
         self.raw_resp.raise_for_status()
+
+        # Also handle 301, 302, and 304
```

Comment:
> By default Requests will perform location redirection for all verbs except HEAD.

Which means that we do not handle 301, 302, 307 and 308 here because requests already take care of them.

Which leads us with 303, 304 and 305 from which Firefox Sync uses 304 only.